database/migrations/2021_12_12_174801_create_order_product_toppings_table.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateOrderProductToppingsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('order_product_toppings', function (Blueprint $table) {
+            $table->foreignId('order_product_id')
+                ->references('id')->on('order_products')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
+            $table->bigInteger('product_id')->unsigned()->nullable(true);
```

Comment:
why it should be nullable?

Reply:
Answer mostly similar to the one for order_products_table.

To be able to delete a product from database without needing to delete all information related to order_products or order_product_toppings.
Example: if we delete an order_product_topping_id from order_product_toppings table because product_id got deleted, order_product_topping_id gets NULL value, but we are still left with an number of toppings ordered for a specific product_id thus means we could focus on adding more toppings for that product in the future.
On the other hand if we use restrictOnDelete we won't be able to delete any products thus making too much redundancy in our products table.

Reply:
What value do you have for removal of the product from database. So you will know that for 400 euro in order were 100 pizza and 50 additional meat for removed Cezar salade?